components/electric/lib/electric/ddlx/command/common.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Electric.DDLX.Command.Common do
+  def sql_repr(value) when value == nil do
+    "null"
+  end
+
+  def sql_repr(value) when is_binary(value) do
+    "'#{escape_quotes(value, [])}'"
+  end
+
+  def sql_repr(values) when is_list(values) do
+    [
+      "ARRAY[",
+      values |> Stream.map(&sql_repr/1) |> Enum.intersperse(", "),
+      "]"
+    ]
+    |> IO.iodata_to_binary()
```

Comment:
Since the whole result is converted to binary in the end, I think something like this would be more efficient:
```suggestion
    array_str = for val <- values, into: "ARRAY[", do: sql_repr(val) <> ", "
    # cut the trailing comma off before appending the closing bracket
    :binary.part(array_str, 0, byte_size(array_str) - 2) <> "]"
```

Reply:
speed is not critical for this, think the existing version is more readable tbh

Reply:
Speed is never critical until someone starts throwing production workloads at it. But readability is important so I'll give you a pass on this ;)